tests/test_outputs/test_form_output_prep.py

```diff
@@ -0,0 +1,328 @@
+"""Tests for 'form_output_prep.py'."""
+# Standard Library Imports
+import pytest
+
+# Third Party Imports
+import pandas as pd
+import numpy as np
+
+# Local Imports
+from src.outputs.form_output_prep import form_output_prep
+
+class TestFormOutputPrep(object):
+    """Tests for form_output_prep."""
+
+    @pytest.fixture(scope="function")
+    def estimated_df(self) -> pd.DataFrame:
+        """'estimated_df' input for form_output_prep.
+        
+        NOTE: This is a subset of columns for testing purposes
+        """
+        columns = [
+            "reference", 
+            "instance", 
+            "period",
+            "201",
+            "202",
+            "211",
+            "602", 
+            "604", 
+            "a_weight", 
+            "formtype",
+            "pg_numeric"
+        ]
+        data = [
+            [39900000404, 0, 202212, 'AA', np.nan, np.nan, 100.0, 'Yes', 6.0, 6, 40],
+            [39900000404, 1, 202212, 'AA', 3146383.0, 4628363.0, 100.0, 'Yes', 6.0, 6, 40],
+            [39900000404, 2, 202212, 'AA', 0.0, 0.0, 100.0, 'Yes', 6.0, 6, 40],
+            [39900000408, 0, 202212, 'AA', np.nan, np.nan, 100.0, 'No', 1.0, 6, 40],
+            [39900000408, 1, 202212, 'AA', 0.0, 1222.0, 100.0, 'No', 1.0, 6, 40], # 604 No and 211 > 0
+            [39900000408, 2, 202212, 'AA', 0.0, 0.0, 100.0, 'No', 1.0, 6, 40],
+            [39900000576, 1, 202212, 'AA', 0.0, 0.0, 100.0, np.nan, 1.0, 1, 40],
+            [39900000960, 0, 202212, 'AA', np.nan, np.nan, np.nan, 'No', 1.0, 1, 40],
+            [39900000960, 1, 202212, 'AA', np.nan, 0.0, np.nan, 'No', 1.0, 1, 40],
+            [39900001029, 1, 202212, 'I', 119376.5, 244543.0, 100.0, np.nan, 1.0, 1, 14]
```

Comment:
A minor change again throughout, "formtype" is always a string, "0001", "0003", "0006" for long forms, NI and short forms respectively. Please could you update all your tests? Sorry I didn't spot this before.